src/test/java/org/opentripplanner/middleware/cdp/ConnectedDataPlatformTest.java

```diff
@@ -0,0 +1,241 @@
+     */
+    @Test
+    public void canCreateZipFileWithContent() throws Exception {
+        assumeTrue(IS_END_TO_END);
+        ConnectedDataManager.IS_TEST = true;
+
+        String userId = UUID.randomUUID().toString();
+        Date startOfYesterday = getStartOfDay(getDateMinusNumberOfDays(new Date(), 1));
+        tripRequest = PersistenceTestUtils.createTripRequest(userId, startOfYesterday);
+        tripSummary = PersistenceTestUtils.createTripSummary(tripRequest.id, startOfYesterday);
+        ConnectedDataManager.stageUploadDays();
+        ConnectedDataManager.processTripHistory();
+        zipFileName = getFileName(startOfYesterday, ConnectedDataManager.ZIP_FILE_NAME_SUFFIX);
+        tempFile = String.join(
+            "/",
+            FileUtils.getTempDirectory().getAbsolutePath(),
+            zipFileName
+        );
+        String fileContents = getContentsOfFileInZip(
+            tempFile,
+            getFileName(startOfYesterday, ConnectedDataManager.DATA_FILE_NAME_SUFFIX)
+        );
+        String contents = ConnectedDataManager.getAnonymizedTripHistory(startOfYesterday, getEndOfDay(startOfYesterday));
+        assertEquals(fileContents, contents);
+    }
+
+    /**
+     * Create a user with trip data and confirm this is written to file. Then remove this user's trip data and confirm
+     * the file is overwritten minus the user's trip data.
+     */
+    @Test
+    public void canRemoveUsersTripDataFromFile() throws Exception {
+        assumeTrue(IS_END_TO_END);
+        ConnectedDataManager.IS_TEST = true;
+
+        String userId = UUID.randomUUID().toString();
+        String batchId = UUID.randomUUID().toString();
+        Date startOfYesterday = getStartOfDay(getDateMinusNumberOfDays(new Date(), 1));
+        tripRequest = PersistenceTestUtils.createTripRequest(userId, batchId, startOfYesterday);
+        tripSummary = PersistenceTestUtils.createTripSummary(tripRequest.id, startOfYesterday);
+        ConnectedDataManager.stageUploadDays();
+        ConnectedDataManager.processTripHistory();
+        zipFileName = getFileName(startOfYesterday, ConnectedDataManager.ZIP_FILE_NAME_SUFFIX);
+        tempFile = String.join(
+            "/",
+            FileUtils.getTempDirectory().getAbsolutePath(),
+            zipFileName
+        );
+        String fileContents = getContentsOfFileInZip(
+            tempFile,
+            getFileName(startOfYesterday, ConnectedDataManager.DATA_FILE_NAME_SUFFIX)
+        );
+        TripHistory tripHistory = JsonUtils.getPOJOFromJSON(fileContents, TripHistory.class);
+        assertTrue(tripHistory.tripRequests.stream().anyMatch(tripRequest -> tripRequest.batchId.equals(batchId)));
+        ConnectedDataManager.removeUsersTripHistory(userId);
+        // Trip request and summary are removed as part of the 'removeUsersTripHistory' method. Setting these to null
+        // prevents a delete error under the tidy-up process.
+        tripRequest = null;
+        tripSummary = null;
```

Comment:
Test cases shouldn't have side effects on other test cases if at all possible. Is there a way these variables can be made to have a scope active only within this method?